src/addons/par/icc/tools/icc_common.sh

```diff
@@ -58,6 +58,11 @@ done
 
 set -ex
 
+if [[ "$script_dir" == "" ]]; then
+    echo "script_dir not set"
+    exit 1
+fi
+
```

Comment:
Don't you need to put script_dir in the PATH or something? Later you call readlink_array with no script_dir prefix and I don't understand how that will be available.